test/jdk/java/lang/System/patchedModule/ModulePatcherTest.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2024, Red Hat, Inc.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.lang.reflect.Constructor;
+
+/*
+ * @test id=unpatched
+ * @summary Test property jdk.patched for unpatched runtime
+ * @run main/othervm ModulePatcherTest false
+ */
+
+/*
+ * @test id=patched
+ * @summary Test property jdk.patched for patched runtime
+ * @compile --patch-module java.base=${test.src}/patch/java/lang
+ *          ${test.src}/patch/java/lang/TestInteger.java ModulePatcherTest.java
+ * @run main/othervm --patch-module=java.base=${test.classes} ModulePatcherTest true
+ */
+public class ModulePatcherTest {
```

Comment:
The existing tests for --patch-module are in test/jdk/tools/launcher/modules/patch, we might want to see if we should just update one of the existing tests or move the new test with a name that makes it clear it's testing the property.